# University of Michigan DEI Information Analysis for 2017 
## Data Manipulation and Preliminary Analyses

### Problem/Opportunity Statement:
Examine 5 academic years from NCAA data page and DEI 2017 conference excel sheet. Determine any trends and other relevant analysis that’s informative of the trends. 


### Project Deliverables: 
* Create a centralized DataFrame of  DEI excel
* Create a few  DataFrames from NCAA data page to compare with DEI 2017 conference excel sheet
     * http://web1.ncaa.org/rgdSearch/exec/saSearch
* Create visualizations to display relevant trends showing racial disparities between conferences and academic years 
    * possible additonal tools: https://us-east-1.online.tableau.com/#/site/visualizationpracticetest01/workbooks



Assumptions 
* Race and ethnicity categorizes minority as nonwhite
* Diversity measured by the proportion of minority students to the whole population.  
* All conferences includes: Michigan, Big 10, ACC, SEC, PAC, Big 12


# Import Libraries 

In [9]:
import pandas as pd

import seaborn as sns
import numpy as np
import matplotlib.pyplot as plte

import plotly.plotly as py
import plotly.graph_objs as go


## Michigan Student Athlete survey Data

Load & Read File 

In [10]:
filename= "2017_DEI_Student Athlete Conference Breakdown.xlsx"

In [11]:
df4=pd.read_excel(filename,sheet_name=2)

# Clean Data 
* Rename columns 
* drop rows and columns 

In [12]:
michigan=(df4.iloc[:30]).copy()

michigan= michigan.rename(index=str, columns={"2016-2017 Michigan Athletics Academic Year (Figures)": "Sport","Unnamed: 1": "White", "Unnamed: 2": "Black",
                                          "Unnamed: 3": "American Indian/Alaskan Native", "Unnamed: 4": "Asian",
                                         "Unnamed: 5": "Hispanic/Latino", "Unnamed: 6": "Native Hawaiian/Pacific Islander",
                                         "Unnamed: 7": "Two or More Races", "Unnamed: 8": "Nonresident Alien",
                                         "Unnamed: 9": "Other", "Unnamed: 10": "Total"})
# michigan.head()
michigan= michigan.set_index("Sport")

In [13]:
michigan= michigan.drop("Sport",0)
michigan= michigan.drop("Total",1)
michigan.head()

White Black American Indian/Alaskan Native Asian  \
Sport                                                                    
Baseball                 29     4                              0     0   
Men's Basketball          8     6                              0     0   
Women's Basketball        4     5                              0     0   
Men's Cross Country      24     0                              0     0   
Women's Cross Country    39     0                              0     0   

                      Hispanic/Latino Native Hawaiian/Pacific Islander  \
Sport                                                                    
Baseball                            1                                0   
Men's Basketball                    0                                0   
Women's Basketball                  1                                0   
Men's Cross Country                 0                                0   
Women's Cross Country               0                                0   

                      Two or More Races Nonresident Alien Other  
Sport                                                            
Baseball                              0                 0     2  
Men's Basketball                      2                 1     0  
Women's Basketball                    4                 0     1  
Men's Cross Country                   0                 2     1  
Women's Cross Country                 0                 1     0

In [14]:
# Change values from objects to float 
michigan.dtypes

White                               object
Black                               object
American Indian/Alaskan Native      object
Asian                               object
Hispanic/Latino                     object
Native Hawaiian/Pacific Islander    object
Two or More Races                   object
Nonresident Alien                   object
Other                               object
dtype: object

In [15]:
michigan=michigan.astype('float64')
michigan.dtypes

White                               float64
Black                               float64
American Indian/Alaskan Native      float64
Asian                               float64
Hispanic/Latino                     float64
Native Hawaiian/Pacific Islander    float64
Two or More Races                   float64
Nonresident Alien                   float64
Other                               float64
dtype: object

## Describe & Summerize Michigan data

In [16]:
numeric_summary = michigan.describe()
size =michigan.shape

print ("rows & columns:",size)
print (numeric_summary)

rows & columns: (29, 9)
           White      Black  American Indian/Alaskan Native      Asian  \
count  29.000000  29.000000                            29.0  29.000000   
mean   26.586207   4.103448                             0.0   0.793103   
std    19.256756   8.295200                             0.0   1.264327   
min     4.000000   0.000000                             0.0   0.000000   
25%    13.000000   0.000000                             0.0   0.000000   
50%    24.000000   1.000000                             0.0   0.000000   
75%    35.000000   4.000000                             0.0   1.000000   
max    82.000000  44.000000                             0.0   5.000000   

       Hispanic/Latino  Native Hawaiian/Pacific Islander  Two or More Races  \
count        29.000000                         29.000000          29.000000   
mean          1.172414                          0.103448           1.482759   
std           1.255530                          0.557086           2.680

There are on average 26 white players per team and 4 or less minorities on average per team, with Native Indian, Asain and Pacific Islander accounting for less that 1 or 0. 

## Analyzing Minority data: what percent of Michigan athetes are minority? 

In [17]:
minority_df=michigan[['Black','American Indian/Alaskan Native','Asian','Hispanic/Latino','Native Hawaiian/Pacific Islander','Two or More Races','Nonresident Alien','Other']]

sorted_sum= ((minority_df.sum()).sort_values(ascending =False))
um_total_minority= sorted_sum.sum()

# sorted_sum
print('total minority:')
print (um_total_minority)
um_total_students= michigan.sum().sum()
print('total student athletes:')
print (um_total_students)

total minority:
324.0
total student athletes:
1095.0


In [18]:
minority_df.head()

Black  American Indian/Alaskan Native  Asian  \
Sport                                                                 
Baseball                 4.0                             0.0    0.0   
Men's Basketball         6.0                             0.0    0.0   
Women's Basketball       5.0                             0.0    0.0   
Men's Cross Country      0.0                             0.0    0.0   
Women's Cross Country    0.0                             0.0    0.0   

                       Hispanic/Latino  Native Hawaiian/Pacific Islander  \
Sport                                                                      
Baseball                           1.0                               0.0   
Men's Basketball                   0.0                               0.0   
Women's Basketball                 1.0                               0.0   
Men's Cross Country                0.0                               0.0   
Women's Cross Country              0.0                               0.0   

                       Two or More Races  Nonresident Alien  Other  
Sport                                                               
Baseball                             0.0                0.0    2.0  
Men's Basketball                     2.0                1.0    0.0  
Women's Basketball                   4.0                0.0    1.0  
Men's Cross Country                  0.0                2.0    1.0  
Women's Cross Country                0.0                1.0    0.0

In [19]:
minority_value= (um_total_minority/um_total_students)
majority_value = 1- (minority_value)

print ('This is the percent of the minority population for Michigan Athletics: ')
print(minority_value)
print ('This is the percent of the majority population for Michigan Athletics: ')
print(majority_value)

This is the percent of the minority population for Michigan Athletics: 
0.2958904109589041
This is the percent of the majority population for Michigan Athletics: 
0.7041095890410959


29.6% of michigan athletes are minority 

## All Conferences: Big-12, SEC, ACC, PAC-12, Big Ten 
* load,clean, and summerize  data from http://web1.ncaa.org/rgdSearch/exec/saSearch

Load & Read excel File

In [20]:
df=pd.read_excel(filename,sheet_name=0)

In [21]:
#rename columns 
df2= df.rename(index=str, columns={"2016-2017 Big Ten Academic Year (Figures)": "Sport","Unnamed: 1": "White", "Unnamed: 2": "Black",
                                          "Unnamed: 3": "American Indian/Alaskan Native", "Unnamed: 4": "Asian",
                                         "Unnamed: 5": "Hispanic/Latino", "Unnamed: 6": "Native Hawaiian/Pacific Islander",
                                         "Unnamed: 7": "Two or More Races", "Unnamed: 8": "Nonresident Alien",
                                         "Unnamed: 9": "Other", "Unnamed: 10": "Total",
                                         "Unnamed: 13": "White %", "Unnamed: 14": "Black %",
                                         "Unnamed: 15": "American Indian/Alaskan Native %", "Unnamed: 16": "Asian %",
                                         "Unnamed: 17": "Hispanic/Latino %", "Unnamed: 18": "Native Hawaiian/Pacific Islander %",
                                         "Unnamed: 19": "Two or More Races %", "Unnamed: 20": "Nonresident Alien %",
                                         "Unnamed: 21": "Other %", "Unnamed: 22": "Total %"})

In [22]:
# add column for conference 

df2['conferences']=df2.shape[0]*['Big Ten']

# change conference column values for each conference 

df2.loc[33:63, 'conferences']= 'Pac12_AcdemYr'
df2.loc[67:97, 'conferences']= 'SEC_AcdemYr'
df2.loc[100:130, 'conferences']= 'ACC_AcdemYr'
df2.loc[133:163, 'conferences']= 'Big12_AcdemYr'


drop columns & rows 

In [23]:
df2= df2.set_index("Sport")
df2= df2.drop("Sport",0)
df2=df2.drop('2016-2017 Big Ten Academic Year (%)',1)
df2=df2.drop("Unnamed: 11",1)

# Looking at each conference: 
* cleaning data  by conference   

In [26]:
BigTen = (df2.iloc[:29]).copy()
# BigTen.head()

In [27]:
#drop columns and rows 
BigTen = BigTen.drop("Total",1)

BigTen = BigTen.drop("Total %",1)

BigTen = BigTen.drop("conferences",1)

In [28]:
#change values from object to float 
BigTen=BigTen.astype('float64')
BigTen.dtypes 

White                                 float64
Black                                 float64
American Indian/Alaskan Native        float64
Asian                                 float64
Hispanic/Latino                       float64
Native Hawaiian/Pacific Islander      float64
Two or More Races                     float64
Nonresident Alien                     float64
Other                                 float64
White %                               float64
Black %                               float64
American Indian/Alaskan Native %      float64
Asian %                               float64
Hispanic/Latino %                     float64
Native Hawaiian/Pacific Islander %    float64
Two or More Races %                   float64
Nonresident Alien %                   float64
Other %                               float64
dtype: object

In [29]:
Pac12 =(df2.iloc[33:62]).copy()

# Pac12

In [30]:
#drop columns and rows
Pac12 = Pac12.drop("Total",1)

Pac12 = Pac12.drop("Total %",1)

Pac12 = Pac12.drop("conferences",1)

In [31]:
#change values from objects to float 
Pac12=Pac12.astype('float64')

Pac12.dtypes

White                                 float64
Black                                 float64
American Indian/Alaskan Native        float64
Asian                                 float64
Hispanic/Latino                       float64
Native Hawaiian/Pacific Islander      float64
Two or More Races                     float64
Nonresident Alien                     float64
Other                                 float64
White %                               float64
Black %                               float64
American Indian/Alaskan Native %      float64
Asian %                               float64
Hispanic/Latino %                     float64
Native Hawaiian/Pacific Islander %    float64
Two or More Races %                   float64
Nonresident Alien %                   float64
Other %                               float64
dtype: object

In [32]:
SEC = (df2.iloc[65:94]).copy()
# SEC


In [33]:
#drop columns and rows 
SEC = SEC.drop("Total",1)

SEC = SEC.drop("Total %",1)

SEC = SEC.drop("conferences",1)

In [34]:
#change values from objects to float 
SEC=SEC.astype('float64')

SEC.dtypes

White                                 float64
Black                                 float64
American Indian/Alaskan Native        float64
Asian                                 float64
Hispanic/Latino                       float64
Native Hawaiian/Pacific Islander      float64
Two or More Races                     float64
Nonresident Alien                     float64
Other                                 float64
White %                               float64
Black %                               float64
American Indian/Alaskan Native %      float64
Asian %                               float64
Hispanic/Latino %                     float64
Native Hawaiian/Pacific Islander %    float64
Two or More Races %                   float64
Nonresident Alien %                   float64
Other %                               float64
dtype: object

In [35]:
ACC = (df2.iloc[97:126]).copy()
# ACC

In [36]:
#drop columns and rows 
ACC  = ACC .drop("Total",1)

ACC  = ACC .drop("Total %",1)

ACC  = ACC .drop("conferences",1)

In [37]:
#change values from objects to float 
ACC=ACC.astype('float64')

ACC.dtypes


White                                 float64
Black                                 float64
American Indian/Alaskan Native        float64
Asian                                 float64
Hispanic/Latino                       float64
Native Hawaiian/Pacific Islander      float64
Two or More Races                     float64
Nonresident Alien                     float64
Other                                 float64
White %                               float64
Black %                               float64
American Indian/Alaskan Native %      float64
Asian %                               float64
Hispanic/Latino %                     float64
Native Hawaiian/Pacific Islander %    float64
Two or More Races %                   float64
Nonresident Alien %                   float64
Other %                               float64
dtype: object

In [38]:
Big12 = (df2.iloc[129:163]).copy()
# Big12

In [39]:
#drop columns and rows 
Big12  = Big12 .drop("Total",1)

Big12 = Big12.drop("Total %",1)

Big12= Big12.drop("conferences",1)

In [40]:
#change vlaues from objects to float 
Big12=Big12.astype('float64')

Big12.dtypes

White                                 float64
Black                                 float64
American Indian/Alaskan Native        float64
Asian                                 float64
Hispanic/Latino                       float64
Native Hawaiian/Pacific Islander      float64
Two or More Races                     float64
Nonresident Alien                     float64
Other                                 float64
White %                               float64
Black %                               float64
American Indian/Alaskan Native %      float64
Asian %                               float64
Hispanic/Latino %                     float64
Native Hawaiian/Pacific Islander %    float64
Two or More Races %                   float64
Nonresident Alien %                   float64
Other %                               float64
dtype: object

## Decribing the data for each conference 

In [42]:
#  BIG TEN 


numeric_summary = BigTen.describe()
size =BigTen.shape
print('BIGTEN')
print ("rows & columns:",size)
print (numeric_summary)

BIGTEN
rows & columns: (29, 18)
            White       Black  American Indian/Alaskan Native      Asian  \
count   29.000000   29.000000                       29.000000  29.000000   
mean   253.344828   58.758621                        0.655172   5.103448   
std    181.839269  132.344155                        1.142549   6.410621   
min     32.000000    0.000000                        0.000000   0.000000   
25%    112.000000    5.000000                        0.000000   2.000000   
50%    191.000000   12.000000                        0.000000   3.000000   
75%    356.000000   34.000000                        1.000000   6.000000   
max    761.000000  696.000000                        4.000000  25.000000   

       Hispanic/Latino  Native Hawaiian/Pacific Islander  Two or More Races  \
count        29.000000                         29.000000          29.000000   
mean         12.655172                          0.551724          18.896552   
std          11.094644                        

There is a mean of ~30% minoroity for all sports 

In [43]:
###  Big-10  ###

BTminority_df=BigTen[['Black','American Indian/Alaskan Native','Asian','Hispanic/Latino','Native Hawaiian/Pacific Islander','Two or More Races','Nonresident Alien','Other']]

BTall_students =BigTen[['White', 'Black','American Indian/Alaskan Native','Asian','Hispanic/Latino','Native Hawaiian/Pacific Islander','Two or More Races','Nonresident Alien','Other']]

BTsorted_sum= ((BTminority_df.sum()).sort_values(ascending =False))
BTtotal_minority= BTsorted_sum.sum()

# sorted_sum
print('In the BIG TEN there are: ')
print('total minority student athletes:')
print (BTtotal_minority)
BTtotal_students= BTall_students.sum().sum()
print('total student athletes:')
print (BTtotal_students)


In the BIG TEN there are: 
total minority student athletes:
3571.0
total student athletes:
10918.0


In [44]:
# BIG TEN minority 

BTminority_value= (BTtotal_minority/BTtotal_students)
BTmajority_value = 1- (BTminority_value)

print ('BigTen Athletics minority percentage: ')
print (BTminority_value )
print ('BigTen Athletics majority population: ')
print (BTmajority_value)

BigTen Athletics minority percentage: 
0.3270745557794468
BigTen Athletics majority population: 
0.6729254442205532


In [46]:
# PAC12

numeric_summary = Pac12.describe()
size =Pac12.shape
print('PAC12')
print ("rows & columns:",size)
print (numeric_summary)

PAC12
rows & columns: (29, 18)
            White       Black  American Indian/Alaskan Native      Asian  \
count   29.000000   29.000000                       29.000000  29.000000   
mean   129.206897   40.517241                        1.896552  11.896552   
std    109.465382   93.823398                        2.023872   8.474098   
min      0.000000    0.000000                        0.000000   0.000000   
25%     43.000000    2.000000                        0.000000   5.000000   
50%     91.000000    8.000000                        2.000000  12.000000   
75%    193.000000   23.000000                        3.000000  15.000000   
max    464.000000  495.000000                        8.000000  36.000000   

       Hispanic/Latino  Native Hawaiian/Pacific Islander  Two or More Races  \
count        29.000000                         29.000000          29.000000   
mean         15.620690                          3.965517          14.275862   
std          15.010505                         

mean 49% minority 

In [47]:
###  PAC-12  ###
PACminority_df=Pac12[['Black','American Indian/Alaskan Native','Asian','Hispanic/Latino','Native Hawaiian/Pacific Islander','Two or More Races','Nonresident Alien','Other']]

PACall_students =Pac12[['White', 'Black','American Indian/Alaskan Native','Asian','Hispanic/Latino','Native Hawaiian/Pacific Islander','Two or More Races','Nonresident Alien','Other']]

PACsorted_sum= ((PACminority_df.sum()).sort_values(ascending =False))
PACtotal_minority= PACsorted_sum.sum()

print('In the PAC12 there are: ')
print('total minority student athletes:')
print (PACtotal_minority)
PACtotal_students= PACall_students.sum().sum()
print('total student athletes:')
print (PACtotal_students )

In the PAC12 there are: 
total minority student athletes:
3697.0
total student athletes:
7444.0


In [49]:
# minority 

PACminority_value= (PACtotal_minority/PACtotal_students)
PACmajority_value = 1- (PACminority_value)

print ('PAC12 Athletics minority population: ')
print(PACminority_value )

print ('PAC12 Athletics majority population: ')
print(PACmajority_value)

PAC12 Athletics minority population: 
0.49664159054271895
PAC12 Athletics majority population: 
0.503358409457281


In [50]:
# SEC 


numeric_summary = SEC.describe()
size =SEC.shape
print('SEC')
print ("rows & columns:",size)
print (numeric_summary)

SEC
rows & columns: (29, 18)
            White       Black  American Indian/Alaskan Native      Asian  \
count   29.000000   29.000000                       29.000000  29.000000   
mean   157.137931   76.827586                        1.137931   2.068966   
std    159.166069  188.148131                        1.328913   2.865198   
min      0.000000    0.000000                        0.000000   0.000000   
25%     39.000000    1.000000                        0.000000   0.000000   
50%     87.000000    6.000000                        1.000000   1.000000   
75%    274.000000   39.000000                        2.000000   3.000000   
max    619.000000  984.000000                        5.000000  12.000000   

       Hispanic/Latino  Native Hawaiian/Pacific Islander  Two or More Races  \
count        29.000000                         29.000000          29.000000   
mean          9.344828                          0.931034           8.689655   
std           9.099748                          1

In [51]:
###  SEC  ###

SECminority_df=SEC[['Black','American Indian/Alaskan Native','Asian','Hispanic/Latino','Native Hawaiian/Pacific Islander','Two or More Races','Nonresident Alien','Other']]

SECall_students =SEC[['White', 'Black','American Indian/Alaskan Native','Asian','Hispanic/Latino','Native Hawaiian/Pacific Islander','Two or More Races','Nonresident Alien','Other']]

SECsorted_sum= ((SECminority_df.sum()).sort_values(ascending =False))
SECtotal_minority= SECsorted_sum.sum()

# sorted_sum
# total_minority
SECtotal_students= SECall_students.sum().sum()
SECtotal_students 

print('In the SEC there are: ')
print('total minority student athletes:')
print (SECtotal_minority)
SECtotal_students= SECall_students.sum().sum()
print('total student athletes:')
print (SECtotal_students )

In the SEC there are: 
total minority student athletes:
3549.0
total student athletes:
8106.0


In [52]:
# minority 

SECminority_value= (SECtotal_minority/SECtotal_students)
SECmajority_value = 1- (SECminority_value)

print ('SEC Athletics minority population: ')
print(SECminority_value )

print ('SEC Athletics majority population: ')
print(SECmajority_value)

SEC Athletics minority population: 
0.4378238341968912
SEC Athletics majority population: 
0.5621761658031088


In [54]:
# ACC 


numeric_summary = ACC.describe()
size =ACC.shape
print('ACC')
print ("rows & columns:",size)
print (numeric_summary)

ACC
rows & columns: (29, 18)
            White       Black  American Indian/Alaskan Native      Asian  \
count   29.000000   29.000000                       29.000000  29.000000   
mean   204.620690   71.241379                        0.758621   3.517241   
std    156.496101  168.493250                        1.430663   3.785505   
min      0.000000    0.000000                        0.000000   0.000000   
25%     66.000000    6.000000                        0.000000   0.000000   
50%    182.000000   11.000000                        0.000000   3.000000   
75%    326.000000   53.000000                        1.000000   6.000000   
max    653.000000  896.000000                        5.000000  14.000000   

       Hispanic/Latino  Native Hawaiian/Pacific Islander  Two or More Races  \
count        29.000000                         29.000000          29.000000   
mean         13.068966                          0.689655          15.344828   
std          11.844019                          1

In [55]:
###  ACC   ###

ACCminority_df=ACC[['Black','American Indian/Alaskan Native','Asian','Hispanic/Latino','Native Hawaiian/Pacific Islander','Two or More Races','Nonresident Alien','Other']]

ACCall_students =ACC[['White', 'Black','American Indian/Alaskan Native','Asian','Hispanic/Latino','Native Hawaiian/Pacific Islander','Two or More Races','Nonresident Alien','Other']]

ACCsorted_sum= ((ACCminority_df.sum()).sort_values(ascending =False))
ACCtotal_minority= ACCsorted_sum.sum()



print('total minority student athletes:')
print (ACCtotal_minority)
ACCtotal_students= ACCall_students.sum().sum()
print('total student athletes:')
print (ACCtotal_students )

total minority student athletes:
4068.0
total student athletes:
10002.0


In [56]:
#minority 

ACCminority_value= (ACCtotal_minority/ACCtotal_students)
ACCmajority_value = 1- (ACCminority_value)

print ('ACC Athletics minority population: ')
print(ACCminority_value )

print ('ACC Athletics majority population: ')
print(ACCmajority_value)

ACC Athletics minority population: 
0.40671865626874626
ACC Athletics majority population: 
0.5932813437312537


In [58]:
# BIG12

numeric_summary = Big12.describe()
size =Big12.shape
print('BIG12')
print ("rows & columns:",size)
print (numeric_summary)

BIG12
rows & columns: (29, 18)
            White       Black  American Indian/Alaskan Native      Asian  \
count   29.000000   29.000000                       29.000000  29.000000   
mean   114.517241   47.482759                        1.931034   2.103448   
std    121.514438  117.826331                        2.852706   2.769023   
min      0.000000    0.000000                        0.000000   0.000000   
25%     24.000000    0.000000                        0.000000   0.000000   
50%     84.000000    8.000000                        1.000000   1.000000   
75%    194.000000   24.000000                        2.000000   3.000000   
max    488.000000  621.000000                       11.000000  14.000000   

       Hispanic/Latino  Native Hawaiian/Pacific Islander  Two or More Races  \
count        29.000000                         29.000000          29.000000   
mean          8.689655                          0.620690          10.931034   
std           8.984524                         

In [59]:
###  Big-12  ###


B12minority_df=Big12[['Black','American Indian/Alaskan Native','Asian','Hispanic/Latino','Native Hawaiian/Pacific Islander','Two or More Races','Nonresident Alien','Other']]

B12all_students =Big12[['White', 'Black','American Indian/Alaskan Native','Asian','Hispanic/Latino','Native Hawaiian/Pacific Islander','Two or More Races','Nonresident Alien','Other']]

B12sorted_sum= ((B12minority_df.sum()).sort_values(ascending =False))
B12total_minority= B12sorted_sum.sum()


print('total minority student athletes:')
print (B12total_minority)
B12total_students= B12all_students.sum().sum()
print('total student athletes:')
print (B12total_students )


total minority student athletes:
2552.0
total student athletes:
5873.0


In [60]:
# minority 

B12minority_value= (B12total_minority/B12total_students)
B12majority_value = 1- (B12minority_value)

print ('Big12 Athletics minority population: ')
print(B12minority_value )

print ('Big12 Athletics majority population: ')
print(B12majority_value)

Big12 Athletics minority population: 
0.4345309041375788
Big12 Athletics majority population: 
0.5654690958624212


## Nationally 

In [69]:
#total minority percentage 
#total athletes 

total_allstudents= (B12total_students + ACCtotal_students+ SECtotal_students+ PACtotal_students+ BTtotal_students)
total_allminority= (B12total_minority + ACCtotal_minority + SECtotal_minority +PACtotal_minority + BTtotal_minority)

percent_minority= total_allminority /total_allstudents
total_majority= 1 - percent_minority

print ('Nationally, the minority population percentage is:') 
print(percent_minority)


Nationally, the minority population percentage is:
0.4118036039014713


In [75]:
print('Out of %d Student Athletes playing D1 sports in America, there are %d minority Student Athletes' % (total_allminority, total_allstudents))

Out of 17437 Student Athletes playing D1 sports in America, there are 42343 minority Student Athletes


### what are the sports have the highest diversity and the lowest diverstiy 

In [76]:
umich= michigan.reset_index()
umich['total']= umich.sum(axis=1)
umich['majority_value']=umich['White']/umich['total']
umich['minority_value'] = 1 - umich['majority_value'] 

umich=umich.sort_values(by=['minority_value'],ascending= False)
umich.set_index('Sport')


UMlowdiversity=umich.tail(3)['Sport']
UMmostdiversity=umich.head(3)['Sport']
print('Michigan teams with the lowest diversity: ')
print (UMlowdiversity)
print('Michigan teams with the highest diversity: ')
print(UMmostdiversity)

Michigan teams with the lowest diversity: 
3       Men's Cross Country
16           Women's Soccer
4     Women's Cross Country
Name: Sport, dtype: object
Michigan teams with the highest diversity: 
2    Women's Basketball
1      Men's Basketball
6              Football
Name: Sport, dtype: object


In [89]:
BTdf= BTall_students.reset_index()
BTdf['total']= BTdf.sum(axis=1)

BTdf['majority_value']=BTdf['White']/BTdf['total']
BTdf['minority_value'] = 1 - BTdf['majority_value'] 

BTdf=BTdf.sort_values(by=['minority_value'],ascending= False)
BTdf.set_index('Sport')


BTlowdiversity=list(BTdf.tail(3)['Sport'])
BTmostdiversity=BTdf.head(3)['Sport']
print('Big Ten teams with the lowest diversity: ')
print ( BTlowdiversity)
print('Big Ten teams with the highest diversity: ')
print(BTmostdiversity)

Big Ten teams with the lowest diversity: 
["Women's Lacrosse", "Women's Cross Country", 'Baseball']
Big Ten teams with the highest diversity: 
1      Men's Basketball
2    Women's Basketball
6              Football
Name: Sport, dtype: object


In [78]:
# d1df= total_allstudents.reset_index()
# d1df['total']= d1df.sum(axis=1)

# d1df['majority_value']=d1df['White']/d1df['total']
# d1df['minority_value'] = 1 - d1df['majority_value'] 

# D1df=D1df.sort_values(by=['minority_value'],ascending= False)
# D1df.set_index('Sport')

# d1lowdiversity=D1df.tail(3)['Sport']
# d1mostdiversity=D1df.head(3)['Sport']
# print('For US D1 sports, the sports with the lowest diversity: ')
# print (d1lowdiversity)
# print('For US D1 sports, the sports with the highest diversity: ')
# print(d1mostdiversity)

In [79]:
PACdf= PACall_students.reset_index()
PACdf['total']= PACdf.sum(axis=1)

PACdf['majority_value']=PACdf['White']/PACdf['total']
PACdf['minority_value'] = 1 - PACdf['majority_value'] 

PACdf=PACdf.sort_values(by=['minority_value'],ascending= False)
PACdf = PACdf.drop([12], axis=0)

PACdf.set_index('Sport')

PAClowdiversity=PACdf.tail(3)['Sport']
PACmostdiversity=PACdf.head(3)['Sport']
print('PAC teams with the lowest diversity: ')
print ( PAClowdiversity)
print('PAC teams with the highest diversity: ')
print(PACmostdiversity)

PAC teams with the lowest diversity: 
5                  Field Hockey
19    Women's Swimming & Diving
11                   Ice Hockey
Name: Sport, dtype: object
PAC teams with the highest diversity: 
2    Women's Basketball
8          Women's Golf
1      Men's Basketball
Name: Sport, dtype: object


In [80]:
# PACdf

In [81]:
SECdf= SECall_students.reset_index()
SECdf['total']= SECdf.sum(axis=1)

SECdf['majority_value']=SECdf['White']/SECdf['total']
SECdf['minority_value'] = 1 - SECdf['majority_value'] 

SECdf=SECdf.sort_values(by=['minority_value'],ascending= False)
SECdf = SECdf.drop([5,9,11,12,14,27], axis=0)
SECdf.set_index('Sport')

SEClowdiversity=SECdf.tail(3)['Sport']
SECmostdiversity=SECdf.head(3)['Sport']
print('SEC teams with the lowest diversity: ')
print ( SEClowdiversity)
print('SEC teams with the highest diversity: ')
print(SECmostdiversity)

SEC teams with the lowest diversity: 
4     Women's Cross Country
0                 Basebeall
13         Women's Lacrosse
Name: Sport, dtype: object
SEC teams with the highest diversity: 
2    Women's Basketball
1      Men's Basketball
6              Football
Name: Sport, dtype: object


In [82]:
# SECdf

In [83]:
ACCdf= ACCall_students.reset_index()
ACCdf['total']= ACCdf.sum(axis=1)

ACCdf['majority_value']=ACCdf['White']/ACCdf['total']
ACCdf['minority_value'] = 1 - ACCdf['majority_value'] 

ACCdf=ACCdf.sort_values(by=['minority_value'],ascending= False)
ACCdf = ACCdf.drop([9,11,27], axis=0)

ACCdf.set_index('Sport')

ACClowdiversity=ACCdf.tail(3)['Sport']
ACCmostdiversity=ACCdf.head(3)['Sport']
print('ACC teams with the lowest diversity: ')
print ( ACClowdiversity)
print('ACC teams with the highest diversity: ')
print(ACCmostdiversity)

ACC teams with the lowest diversity: 
0            Basebeall
12      Men's Lacrosse
13    Women's Lacrosse
Name: Sport, dtype: object
ACC teams with the highest diversity: 
2    Women's Basketball
1      Men's Basketball
6              Football
Name: Sport, dtype: object


In [84]:
# ACCdf
# ACCall_students

In [85]:
b12df=B12all_students.reset_index()
b12df['total']= b12df.sum(axis=1)

b12df['majority_value']=b12df['White']/b12df['total']
b12df['minority_value'] = 1 - b12df['majority_value'] 

b12df=b12df.sort_values(by=['minority_value'],ascending= False)
b12df = b12df.drop([5,11,12,13,27], axis=0)
b12df.set_index('Sport')

b12lowdiversity=b12df.tail(3)['Sport']
b12mostdiversity=b12df.head(3)['Sport']
print('Big 12 teams with the lowest diversity: ')
print (b12lowdiversity)
print('Big 12 teams with the highest diversity: ')
print(b12mostdiversity)

Big 12 teams with the lowest diversity: 
14                       Rowing
0                     Basebeall
19    Women's Swimming & Diving
Name: Sport, dtype: object
Big 12 teams with the highest diversity: 
1       Men's Basketball
2     Women's Basketball
21        Women's Tennis
Name: Sport, dtype: object


In [103]:
#BIG10 


BTlowdiversity=', '.join(list(BTdf.tail(3)['Sport']))
BTmostdiversity=', '.join(list(BTdf.head(3)['Sport']))
print('Big Ten teams with the lowest diversity: %s'%(BTlowdiversity))
print ('                           ')
print('Big Ten teams with the highest diversity: %s' %(BTmostdiversity))
print ('-----------------------')
print ('                           ')

#PAC12 


PAClowdiversity= ', '.join(PACdf.tail(3)['Sport'])
PACmostdiversity=', '.join(PACdf.head(3)['Sport'])
print('PAC teams with the lowest diversity: %s '%( PAClowdiversity) )
print ('                           ')
print('PAC teams with the highest diversity: %s' % (PACmostdiversity))
print ('-----------------------')
print ('                           ')


# SEC


SEClowdiversity=', '.join(SECdf.tail(3)['Sport'])
SECmostdiversity=', '.join(SECdf.head(3)['Sport'])
print('SEC teams with the lowest diversity: %s '%( SEClowdiversity) )
print ('                           ')
print('SEC teams with the highest diversity: %s '% (SECmostdiversity))
print ('-----------------------')
print ('                           ')

# ACC

ACClowdiversity=', '.join(ACCdf.tail(3)['Sport'])
ACCmostdiversity=', '.join(ACCdf.head(3)['Sport'])
print('ACC teams with the lowest diversity: %s '% ( ACClowdiversity))
print ('                           ')
print('ACC teams with the highest diversity: %s '% (ACCmostdiversity))
print ('-----------------------')
print ('                           ')


# BIG12

b12lowdiversity=', '.join(b12df.tail(3)['Sport'])
b12mostdiversity=', '.join(b12df.head(3)['Sport'])
print('Big 12 teams with the lowest diversity: %s '% (b12lowdiversity))
print ('                           ')
print('Big 12 teams with the highest diversity:%s '% (b12mostdiversity))
print ('                           ')
print ('                           ')

Big Ten teams with the lowest diversity: Women's Lacrosse, Women's Cross Country, Baseball
                           
Big Ten teams with the highest diversity: Men's Basketball, Women's Basketball, Football
-----------------------
                           
PAC teams with the lowest diversity: Field Hockey, Women's Swimming & Diving, Ice Hockey 
                           
PAC teams with the highest diversity: Women's Basketball, Women's Golf, Men's Basketball
-----------------------
                           
SEC teams with the lowest diversity: Women's Cross Country, Basebeall, Women's Lacrosse 
                           
SEC teams with the highest diversity: Women's Basketball, Men's Basketball, Football 
-----------------------
                           
ACC teams with the lowest diversity: Basebeall, Men's Lacrosse, Women's Lacrosse 
                           
ACC teams with the highest diversity: Women's Basketball, Men's Basketball, Football 
-----------------------
      

The sports with the lowest diversity were different for each conference, however Men's Basketbal and Women's Basketball were among the top 3 sports with the most diversity for every conferences; all except 2 conferences had Football and Men's and  Women's Basketbal for top 3 most diverse. 

# Analyze Data & Interpret Results 
* using plotly package  

## Michign - Minority proportons by Sport 

In [104]:
UMlowdiversity

# host_city=host_city[['Country','Year']]

um_by_sport=umich[['Sport','minority_value','majority_value']]

top_diversity= um_by_sport.iloc[:3]
middle_diversity=um_by_sport.iloc[13:16]
low_diversity=um_by_sport.iloc[26:29]

labels = ['Minority','White']
# values = [minority_value,majority_value

wbball_values=[(top_diversity['minority_value'].iloc[0]),(top_diversity['majority_value'].iloc[0])]
mbball__values=[(top_diversity['minority_value'].iloc[1]),(top_diversity['majority_value'].iloc[1])]
Fball__values= [(top_diversity['minority_value'].iloc[2]),(top_diversity['majority_value'].iloc[2])]

MCC_values=[(low_diversity['minority_value'].iloc[0]),(low_diversity['majority_value'].iloc[0])]
soccer__values=[(low_diversity['minority_value'].iloc[1]),(low_diversity['majority_value'].iloc[1])]
WCC__values= [(low_diversity['minority_value'].iloc[2]),(low_diversity['majority_value'].iloc[2])]

feildhoc_values=[(middle_diversity['minority_value'].iloc[0]),(middle_diversity['majority_value'].iloc[0])]
softball__values=[(middle_diversity['minority_value'].iloc[1]),(middle_diversity['majority_value'].iloc[1])]
wrestle__values= [(middle_diversity['minority_value'].iloc[2]),(middle_diversity['majority_value'].iloc[2])]
print(low_diversity)
print(middle_diversity)


fig = {
    'data': [
        {
            'labels': ['Minority','White'],
            'values': wbball_values,
            'type': 'pie',
            'name': 'Womens Basketball',
            'marker': {'colors': ['rgb(251, 214, 82)',
                                  'rgb(81, 118, 251)']},
            'domain': {'x': [0, 0.25],
                       'y': [0, 0.25]},
            'hoverinfo':'label+percent+name',
            'text': ['Womens Bball'], 
             'textinfo':'text+percent'
        },
        {
            'labels': ['Minority','White'],
            'values': mbball__values,
            'marker': {'colors': ['rgb(251, 214, 82)',
                                  'rgb(81, 118, 251)']},
            'type': 'pie',
            'name': 'Mens Basketball',
            'domain': {'x': [0.38, 0.63],
                       'y': [0, 0.25]},
            'hoverinfo':'label+percent+name',
            'text': ['Mens Bball'], 
             'textinfo':'text+percent'

        },
        {
            'labels': ['Minority','White'],
            'values': Fball__values,
            'marker': {'colors': ['rgb(251, 214, 82)',
                                  'rgb(81, 118, 251)']},
            'type': 'pie',
            'name': 'Football',
            'domain': {'x': [0.75, 1],
                       'y': [0, 0.25]},
            'hoverinfo':'label+percent+name',
            'text': ['Football'], 
             'textinfo':'text+percent'
        },
        {
            'labels': ['Minority','White'],
            'values': feildhoc_values,
            'marker': {'colors': ['rgb(251, 214, 82)',
                                  'rgb(81, 118, 251)']},
            'type': 'pie',
            'name': 'Feild Hockey',
            'domain': {'x': [0, 0.25],
                       'y': [0.38, 0.63]},
            'hoverinfo':'label+percent+name',
            'text': ['Feild Hockey'], 
             'textinfo':'text+percent'
        },
         {
            'labels': ['Minority','White'],
            'values': softball__values,
            'marker': {'colors': ['rgb(251, 214, 82)',
                                  'rgb(81, 118, 251)']},
            'type': 'pie',
            'name': 'Softball',
            'domain': {'x': [0.38, 0.63],
                       'y': [0.38, 0.63]},
            'hoverinfo':'label+percent+name',
             'text': ['Softball'], 
             'textinfo':'text+percent'

        },
         {
            'labels': ['Minority','White'],
            'values': wrestle__values,
            'marker': {'colors': ['rgb(251, 214, 82)',
                                  'rgb(81, 118, 251)']},
            'type': 'pie',
            'name': 'Wrestling',
            'domain': {'x': [0.75, 1],
                       'y': [0.38, 0.63]},
            'hoverinfo':'label+percent+name',
             'text': ['Wrestling'], 
             'textinfo':'text+percent'

        },
        {
            'labels': ['Minority','White'],
            'values': MCC_values,
            'marker': {'colors': ['rgb(251, 214, 82)',
                                  'rgb(81, 118, 251)']},
            'type': 'pie',
            'name': 'Mens Cross Country',
            'domain': {'x': [0, 0.25],
                       'y': [0.70,0.95]},
            'hoverinfo':'label+percent+name',
            'text': ['Mens CrossCountry'],
            'textinfo':'text+percent'

        },
        {
            'labels': ['Minority','White'],
            'values': soccer__values,
            'marker': {'colors': ['rgb(251, 214, 82)',
                                  'rgb(81, 118, 251)']},
            'type': 'pie',
            'name': 'Womens Soccer',
            'domain': {'x': [0.38, 0.63],
                       'y': [0.70, 0.95]},
            'hoverinfo':'label+percent+name',
            'text': ['Womens Soccer'],
            'textinfo':'text+percent'

        },
        {
            'labels': ['Minority','White'],
            'values': WCC__values,
            'marker': {'colors': ['rgb(251, 214, 82)',
                                  'rgb(81, 118, 251)']},
            'type': 'pie',
            'name': 'Womens Cross Country',
            'domain': {'x': [0.75, 1],
                       'y': [0.70, 0.95]},
            'hoverinfo':'label+percent+name',
            'text': ['Womens CrossCountry'],
            'textinfo':'text+percent'

        },

    ],
    'layout': {'title': 'Michigan Minority Proportions By Sport, 2017',
               'showlegend': True,}
    
}

py.iplot(fig, filename='bysport_pie_chart')
# # py.iplot([trace], filename='bysport_pie_chart')
# trace = go.Pie(labels=labels, values=wbball_values)

                    Sport  minority_value  majority_value
3     Men's Cross Country        0.111111        0.888889
16         Women's Soccer        0.096774        0.903226
4   Women's Cross Country        0.025000        0.975000
           Sport  minority_value  majority_value
5   Field Hockey        0.304348        0.695652
17      Softball        0.281250        0.718750
28     Wrestling        0.281250        0.718750


/Applications/anaconda3/lib/python3.6/site-packages/IPython/core/display.py:689: UserWarning:

Consider using IPython.display.IFrame instead



# Big10- Minority proportons by Sport 

In [105]:
(top_diversity['minority_value'].iloc[0]),(top_diversity['majority_value'].iloc[0])

(0.7333333333333334, 0.26666666666666666)

In [106]:
UMlowdiversity,UMmostdiversity,BTlowdiversity,BTmostdiversity

UMlowdiversity

# host_city=host_city[['Country','Year']]

BT_by_sport=BTdf[['Sport','minority_value','majority_value']]

top_diversity= BT_by_sport.iloc[:3]
middle_diversity=BT_by_sport.iloc[13:16]
low_diversity=BT_by_sport.iloc[26:29]

print(top_diversity)
labels = ['Minority','White']

mbball_values=[(top_diversity['minority_value'].iloc[0]),(top_diversity['majority_value'].iloc[0])]
wbball__values=[(top_diversity['minority_value'].iloc[1]),(top_diversity['majority_value'].iloc[1])]
Fball__values= [(top_diversity['minority_value'].iloc[2]),(top_diversity['majority_value'].iloc[2])]

WLax_values=[(low_diversity['minority_value'].iloc[0]),(low_diversity['majority_value'].iloc[0])]
WCC__values=[(low_diversity['minority_value'].iloc[1]),(low_diversity['majority_value'].iloc[1])]
Baseball__values= [(low_diversity['minority_value'].iloc[2]),(low_diversity['majority_value'].iloc[2])]

wopo_values=[(middle_diversity['minority_value'].iloc[0]),(middle_diversity['majority_value'].iloc[0])]
wrestle__values=[(middle_diversity['minority_value'].iloc[1]),(middle_diversity['majority_value'].iloc[1])]
volly__values= [(middle_diversity['minority_value'].iloc[2]),(middle_diversity['majority_value'].iloc[2])]
print(low_diversity)
print(middle_diversity)


fig = {
    'data': [
        {
            'labels': ['Minority','White'],
            'values': wbball_values,
            'type': 'pie',
            'name': 'Womens Basketball',
            'marker': {'colors': ['rgb(251, 214, 82)',
                                  'rgb(81, 118, 251)']},
            'domain': {'x': [0, 0.25],
                       'y': [0, 0.25]},
            'hoverinfo':'label+percent+name',
            'text': ['Womens Bball'], 
             'textinfo':'text+percent'
        },
        {
            'labels': ['Minority','White'],
            'values': mbball__values,
            'marker': {'colors': ['rgb(251, 214, 82)',
                                  'rgb(81, 118, 251)']},
            'type': 'pie',
            'name': 'Mens Basketball',
            'domain': {'x': [0.38, 0.63],
                       'y': [0, 0.25]},
            'hoverinfo':'label+percent+name',
            'text': ['Mens Bball'], 
             'textinfo':'text+percent'

        },
        {
            'labels': ['Minority','White'],
            'values': Fball__values,
            'marker': {'colors': ['rgb(251, 214, 82)',
                                  'rgb(81, 118, 251)']},
            'type': 'pie',
            'name': 'Football',
            'domain': {'x': [0.75, 1],
                       'y': [0, 0.25]},
            'hoverinfo':'label+percent+name',
            'text': ['Football'], 
             'textinfo':'text+percent'
        },
        {
            'labels': ['Minority','White'],
            'values': feildhoc_values,
            'marker': {'colors': ['rgb(251, 214, 82)',
                                  'rgb(81, 118, 251)']},
            'type': 'pie',
            'name': 'Feild Hockey',
            'domain': {'x': [0, 0.25],
                       'y': [0.38, 0.63]},
            'hoverinfo':'label+percent+name',
            'text': ['Feild Hockey'], 
             'textinfo':'text+percent'
        },
         {
            'labels': ['Minority','White'],
            'values': softball__values,
            'marker': {'colors': ['rgb(251, 214, 82)',
                                  'rgb(81, 118, 251)']},
            'type': 'pie',
            'name': 'Softball',
            'domain': {'x': [0.38, 0.63],
                       'y': [0.38, 0.63]},
            'hoverinfo':'label+percent+name',
             'text': ['Softball'], 
             'textinfo':'text+percent'

        },
         {
            'labels': ['Minority','White'],
            'values': wrestle__values,
            'marker': {'colors': ['rgb(251, 214, 82)',
                                  'rgb(81, 118, 251)']},
            'type': 'pie',
            'name': 'Wrestling',
            'domain': {'x': [0.75, 1],
                       'y': [0.38, 0.63]},
            'hoverinfo':'label+percent+name',
             'text': ['Wrestling'], 
             'textinfo':'text+percent'

        },
        {
            'labels': ['Minority','White'],
            'values': MCC_values,
            'marker': {'colors': ['rgb(251, 214, 82)',
                                  'rgb(81, 118, 251)']},
            'type': 'pie',
            'name': 'Mens Cross Country',
            'domain': {'x': [0, 0.25],
                       'y': [0.70,0.95]},
            'hoverinfo':'label+percent+name',
            'text': ['Mens CrossCountry'],
            'textinfo':'text+percent'

        },
        {
            'labels': ['Minority','White'],
            'values': soccer__values,
            'marker': {'colors': ['rgb(251, 214, 82)',
                                  'rgb(81, 118, 251)']},
            'type': 'pie',
            'name': 'Womens Soccer',
            'domain': {'x': [0.38, 0.63],
                       'y': [0.70, 0.95]},
            'hoverinfo':'label+percent+name',
            'text': ['Womens Soccer'],
            'textinfo':'text+percent'

        },
        {
            'labels': ['Minority','White'],
            'values': WCC__values,
            'marker': {'colors': ['rgb(251, 214, 82)',
                                  'rgb(81, 118, 251)']},
            'type': 'pie',
            'name': 'Womens Cross Country',
            'domain': {'x': [0.75, 1],
                       'y': [0.70, 0.95]},
            'hoverinfo':'label+percent+name',
            'text': ['Womens CrossCountry'],
            'textinfo':'text+percent'

        },

    ],
    'layout': {'title': 'Big-10 Minority Proportions By Sport, 2017',
               'showlegend': True,}
    
}

py.iplot(fig, filename='BT_bysport_pie_chart')

                Sport  minority_value  majority_value
1    Men's Basketball        0.668203        0.331797
2  Women's Basketball        0.627451        0.372549
6            Football        0.539625        0.460375
                    Sport  minority_value  majority_value
13       Women's Lacrosse        0.142202        0.857798
4   Women's Cross Country        0.140110        0.859890
0                Baseball        0.132196        0.867804
         Sport  minority_value  majority_value
27  Water Polo        0.255814        0.744186
28   Wrestling        0.253669        0.746331
26  Volleyball        0.253112        0.746888


/Applications/anaconda3/lib/python3.6/site-packages/IPython/core/display.py:689: UserWarning:

Consider using IPython.display.IFrame instead



## Pie chart comparing Michigan's minority proportion to: 
* Pac12, SEC, Big12, Big10, ACC and all conferences 

In [107]:
labels = ['Minority','White']
values = [BTminority_value,BTmajority_value]

trace = go.Pie(labels=labels, values=values)

py.iplot([trace], filename='basic_pie_chart')

/Applications/anaconda3/lib/python3.6/site-packages/IPython/core/display.py:689: UserWarning:

Consider using IPython.display.IFrame instead



In [108]:
labels = ['Minority','White']
values = [percent_minority,total_majority]

trace = go.Pie(labels=labels, values=values)

py.iplot([trace], filename='basic_pie_chart')

/Applications/anaconda3/lib/python3.6/site-packages/IPython/core/display.py:689: UserWarning:

Consider using IPython.display.IFrame instead



* Michigan 3.1% less diverse than Bigten ( 29.6% and 32.7% respectively)
* Michigan is 11.6% less diverse than the total percentage of the national diversity for D! schools.

Michigan only - which sports are more diverse and least diverse

http://www.ncaa.org/about/resources/research/ncaa-demographics-database

In [109]:



fig = {
  "data": [
    {
      "values": [minority_value,majority_value],
      "labels": ['Minority','White'],
        'marker': {'colors': ['rgb(251, 214, 82)',
                                  'rgb(81, 118, 251)']},
         "text":["Michigan"],
      "domain": {"x": [0, 0.25]},
      "name": "Michigan race proportion",
      "hoverinfo":"label+percent+name",
      "hole": .4,
      "type": "pie"
    },
    {
      "values": [BTminority_value,BTmajority_value],
      "labels": ['Minority','White'],
        'marker': {'colors': ['rgb(251, 214, 82)',
                                  'rgb(81, 118, 251)']},
         "text":["BigTen"],
      "domain": {"x": [0.3, 0.55]},
      "name": "Big10 race proportion",
      "hoverinfo":"label+percent+name",
      "hole": .4,
      "type": "pie"
    },
    {
      "values": [percent_minority,total_majority],
      "labels": ['Minority','White'],
        'marker': {'colors': ['rgb(251, 214, 82)',
                                  'rgb(81, 118, 251)']},
      "text":["All Conferences"],
      "textposition":"inside",
      "domain": {"x": [0.65, 0.9]},
      "name": "All conferences race proportion",
      "hoverinfo":"label+percent+name",
      "hole": .4,
      "type": "pie"
    }],
  "layout": {"title":"Student Athlete Race Proportion 2017"}
}
py.iplot(fig, filename='donut')

/Applications/anaconda3/lib/python3.6/site-packages/IPython/core/display.py:689: UserWarning:

Consider using IPython.display.IFrame instead



In [110]:
fig = {
  "data": [
    {
      "values": [minority_value,majority_value],
      "labels": ['Minority','White'],
        'marker': {'colors': ['rgb(251, 214, 82)',
                                  'rgb(81, 118, 251)']},
      "domain": {"x": [0, .48]},
      "name": "Michigan race proportions",
      "hoverinfo":"label+percent+name",
      "hole": .4,
      "type": "pie"
    },
    {
      "values": [percent_minority,total_majority],
      "labels": ['Minority','White'],
        'marker': {'colors': ['rgb(251, 214, 82)',
                                  'rgb(81, 118, 251)']},
#       "text":["All Conferences"],
      "textposition":"inside",
      "domain": {"x": [.52, 1]},
      "name": "All Conferences race proportions",
      "hoverinfo":"label+percent+name",
      "hole": .4,
      "type": "pie"
    }],
  "layout": {
        "title":"Student Athlete Diversity 2017",
        "annotations": [
            {
                "font": {
                    "size": 20
                },
                "showarrow": False,
                "text": "Michigan",
                "x": 0.18,
                "y": 0.5
            },
            {
                "font": {
                    "size": 20
                },
                "showarrow": False,
                "text": "All Conferences",
                "x": 0.85,
                "y": 0.5
            }
        ]
    }
}
py.iplot(fig, filename='donut')

/Applications/anaconda3/lib/python3.6/site-packages/IPython/core/display.py:689: UserWarning:

Consider using IPython.display.IFrame instead



In [111]:


fig = {'data':[{'labels': ['Minority','White'],
               'values': [minority_value,majority_value], 
                'type': 'pie',
               'marker': {'colors': ['rgb(251, 214, 82)',
                                  'rgb(81, 118, 251)']}}],
      'layout': {'title': 'Minority Ratio'}}

py.iplot(fig)


/Applications/anaconda3/lib/python3.6/site-packages/IPython/core/display.py:689: UserWarning:

Consider using IPython.display.IFrame instead



In [112]:
import plotly.plotly as py
import plotly.graph_objs as go

fig = {
    'data': [
        {
            'labels': ['Minority','White'],
            'values': [PACminority_value,PACmajority_value],
            'type': 'pie',
            'name': 'PAC12',
            'marker': {'colors': ['rgb(251, 214, 82)',
                                  'rgb(81, 118, 251)']},
            'domain': {'x': [0, 0.25],
                       'y': [0, 0.45]},
            'hoverinfo':'label+percent+name',
            'textinfo':'value'
        },
        {
            'labels': ['Minority','White'],
            'values': [ACCminority_value,ACCmajority_value],
            'marker': {'colors': ['rgb(251, 214, 82)',
                                  'rgb(81, 118, 251)']},
            'type': 'pie',
            'name': 'ACCs',
            'domain': {'x': [0.3, 0.55],
                       'y': [0, 0.45]},
            'hoverinfo':'label+percent+name',
            'textinfo':'none'

        },
        {
            'labels': ['Minority','White'],
            'values': [SECminority_value,SECmajority_value],
            'marker': {'colors': ['rgb(251, 214, 82)',
                                  'rgb(81, 118, 251)']},
            'type': 'pie',
            'name': 'SEC',
            'domain': {'x': [0, 0.25],
                       'y': [0.55, 1]},
            'hoverinfo':'label+percent+name',
            'textinfo':'none'
        },
        {
            'labels': ['Minority','White'],
            'values': [B12minority_value,B12majority_value],
            'marker': {'colors': ['rgb(251, 214, 82)',
                                  'rgb(81, 118, 251)']},
            'type': 'pie',
            'name': 'Big12',
            'domain': {'x': [0.3, 0.55],
                       'y': [0.55, 1]},
            'hoverinfo':'label+percent+name',
            'textinfo':'none'
        },
         {
            'labels': ['Minority','White'],
            'values': [BTminority_value,BTmajority_value],
            'marker': {'colors': ['rgb(251, 214, 82)',
                                  'rgb(81, 118, 251)']},
            'type': 'pie',
            'name': 'B10',
            'domain': {'x': [0.65, 0.9],
                       'y': [0.55, 1]},
            'hoverinfo':'label+percent+name',
            'textinfo':'none'

        },
         {
            'labels': ['Minority','White'],
            'values': [percent_minority,total_majority],
            'marker': {'colors': ['rgb(251, 214, 82)',
                                  'rgb(81, 118, 251)']},
            'type': 'pie',
            'name': 'All Conferences',
            'domain': {'x': [0.65, 0.9],
                       'y': [0, 0.45]},
            'hoverinfo':'label+percent+name',
            'textinfo':'none'

        },

    ],
    'layout': {'title': 'Diversity Proportions',
               'showlegend': True,
               "annotations":[
                   {
                       "font": {
                    "size": 20
                },
                "showarrow": False,
                "text": "SEC",
                "x": 0.30,
                "y": 0.6
                   },
                   {
                        "font": {
                    "size": 20
                },
                "showarrow": False,
                "text": "SEC",
                "x": 0.30,
                "y": 0.6
                   },
                   {
                        "font": {
                    "size": 20
                },
                "showarrow": False,
                "text": "SEC",
                "x": 0.30,
                "y": 0.6
                   },
                   {
                        "font": {
                    "size": 20
                },
                "showarrow": False,
                "text": "PAC12",
                "x": 0.30,
                "y": 0.1
                   },
                   {
                        "font": {
                    "size": 20
                },
                "showarrow": False,
                "text": "BIG12",
                "x": 0.7,
                "y": 0.6
                   },
                   {
                        "font": {
                    "size": 20
                },
                "showarrow": False,
                "text": "ACC",
                "x": 0.7,
                "y": 0.1
                   }
                   
               ]
              }
    
}

py.iplot(fig, filename='pie_chart_subplots')

/Applications/anaconda3/lib/python3.6/site-packages/IPython/core/display.py:689: UserWarning:

Consider using IPython.display.IFrame instead



In [113]:
fig = {
    'data': [
        {
            'labels': ['Minority','White'],
            'values': [PACminority_value,PACmajority_value],
            'type': 'pie',
            'name': 'PAC12',
            'marker': {'colors': ['rgb(251, 214, 82)',
                                  'rgb(81, 118, 251)']},
            'domain': {'x': [0, 0.25],
                       'y': [0, 0.25]},
            'hoverinfo':'label+percent+name',
            'textinfo':'percent',
            #'text': 'Pace-12',
            'textinfo':'percent'
        },
        {
            'labels': ['Minority','White'],
            'values': [ACCminority_value,ACCmajority_value],
            'marker': {'colors': ['rgb(251, 214, 82)',
                                  'rgb(81, 118, 251)']},
            'type': 'pie',
            'name': 'ACCs',
            'domain': {'x': [0.38, 0.63],
                       'y': [0, 0.25]},
            'hoverinfo':'label+percent+name',
            'textinfo':'percent'

        },
        {
            'labels': ['Minority','White'],
            'values': [SECminority_value,SECmajority_value],
            'marker': {'colors': ['rgb(251, 214, 82)',
                                  'rgb(81, 118, 251)']},
            'type': 'pie',
            'name': 'SEC',
            'domain': {'x': [0, 0.25],
                       'y': [0.38, 0.63]},
            'hoverinfo':'label+percent+name',
            'textinfo':'percent'
        },
        {
            'labels': ['Minority','White'],
            'values': [B12minority_value,B12majority_value],
            'marker': {'colors': ['rgb(251, 214, 82)',
                                  'rgb(81, 118, 251)']},
            'type': 'pie',
            'name': 'Big12',
            'domain': {'x': [0.38, 0.63],
                       'y': [0.38, 0.63]},
            'hoverinfo':'label+percent+name',
            'textinfo':'percent'
        },
         {
            'labels': ['Minority','White'],
            'values': [BTminority_value,BTmajority_value],
            'marker': {'colors': ['rgb(251, 214, 82)',
                                  'rgb(81, 118, 251)']},
            'type': 'pie',
            'name': 'B10',
            'domain': {'x': [0.75, 1],
                       'y': [0.38, 0.63]},
            'hoverinfo':'label+percent+name',
            'textinfo':'percent'

        },
         {
            'labels': ['Minority','White'],
            'values': [percent_minority,total_majority],
            'marker': {'colors': ['rgb(251, 214, 82)',
                                  'rgb(81, 118, 251)']},
            'type': 'pie',
            'name': 'All Conferences',
            'domain': {'x': [0.75, 1],
                       'y': [0, 0.25]},
            'hoverinfo':'label+percent+name',
            'textinfo':'percent'

        },
        {
            'labels': ['Minority','White'],
            'values': [minority_value,majority_value],
            'marker': {'colors': ['rgb(251, 214, 82)',
                                  'rgb(81, 118, 251)']},
            'type': 'pie',
            'name': 'Michigan',
            'domain': {'x': [0.38, 0.63],
                       'y': [0.75, 1]},
            'hoverinfo':'label+percent+name',
            'textinfo':'percent'

        },

    ],
    'layout': {'title': 'Minority Proportions 2017',
               'showlegend': True,
              }
    
}

py.iplot(fig, filename='pie_chart_subplots')

/Applications/anaconda3/lib/python3.6/site-packages/IPython/core/display.py:689: UserWarning:

Consider using IPython.display.IFrame instead



In [114]:
import plotly.plotly as py
import cufflinks as cf
import pandas as pd
import numpy as np
print (cf.__version__)

0.14.6


In [115]:
x=['Big Ten','ACC','SEC','PAC','Big12']
y=[BTtotal_minority,ACCtotal_minority,SECtotal_minority,PACtotal_minority,B12total_minority]
z=[BTtotal_students,ACCtotal_students,SECtotal_students,PACtotal_students,B12total_students]

trace0= go.Bar(x=x,y=y,name='Minority')
trace1= go.Bar(x=x,y=z,name='All Students')

data = [trace0,trace1]
layout= dict(title='Diversity Ratio All conferences',xaxis= dict(title='x-axis'),yaxis= dict(title='y-axis'))




In [116]:
cf.set_config_file(offline=False, world_readable=True, theme='ggplot')

# df = pd.DataFrame(np.random.rand(10, 4), columns=['A', 'B', 'C', 'D'])
michigan.iplot(kind='bar')#, filename='cufflinks/grouped-bar-chart')

/Applications/anaconda3/lib/python3.6/site-packages/IPython/core/display.py:689: UserWarning:

Consider using IPython.display.IFrame instead



In [117]:
data = [['Michigan', minority_value, majority_value],['BigTen',BTminority_value,BTmajority_value],['All conferences',percent_minority,total_majority]]
d2 = pd.DataFrame(data,columns=['Group','Minority', 'White'])
d2

Group  Minority     White
0         Michigan  0.295890  0.704110
1           BigTen  0.327075  0.672925
2  All conferences  0.411804  0.588196

In [118]:
# [[BTtotal_minority,BTtotal_students],[total_allminority,total_allstudents],[um_total_students,um_total_minority]]
x=['Michigan', 'Big Ten', 'All Conferences']
y=[um_total_minority,BTtotal_minority,total_allminority]
z=[um_total_students,BTtotal_students,total_allstudents]

trace0= go.Bar(x=x,y=y,name='Minority')
trace1= go.Bar(x=x,y=z,name='All Students')

data = [trace0,trace1]
layout= dict(title='Diversity Ratio All Conferences',xaxis= dict(title='Conference'),yaxis= dict(title='Number of Student-Athletes'))


fig = dict(data=data , layout=layout)
py.iplot(fig, filename='Student-population')


/Applications/anaconda3/lib/python3.6/site-packages/IPython/core/display.py:689: UserWarning:

Consider using IPython.display.IFrame instead



In [119]:
x=['Big Ten','ACC','SEC','PAC','Big12']
y=[BTtotal_minority,ACCtotal_minority,SECtotal_minority,PACtotal_minority,B12total_minority]
z=[BTtotal_students,ACCtotal_students,SECtotal_students,PACtotal_students,B12total_students]

trace0= go.Bar(x=x,y=y,name='Minority')
trace1= go.Bar(x=x,y=z,name='All Students')

data = [trace0,trace1]
layout= dict(title='Diversity Ratio All Conferences',xaxis= dict(title='Conference'),yaxis= dict(title='Number of Student-Athletes'))


fig = dict(data=data , layout=layout)
py.iplot(fig, filename='Student-minority-population')


/Applications/anaconda3/lib/python3.6/site-packages/IPython/core/display.py:689: UserWarning:

Consider using IPython.display.IFrame instead



In [120]:
# UMlowdiversity=umich.tail(3)['Sport']
# UMmostdiversity=umich.head(3)['Sport']
# BTlowdiversity=BTdf.tail(3)['Sport']
# BTmostdiversity=BTdf.head(3)['Sport']

um_bysport= michigan.reset_index()
um_bysport['total']= umich.sum(axis=1)
um_bysport['majority_value']=um_bysport['White']/um_bysport['total']
um_bysport['minority_value'] = 1 - um_bysport['majority_value'] 



data = [go.Bar(x=umich.Sport,
            y=umich.minority_value)]
layout= dict(title='Michigan Race Breakdown by Sport',yaxis= dict(title='Minority Proportion'))


fig = dict(data=data , layout=layout)

py.iplot(fig, filenbame='Michigan_by_sport')

/Applications/anaconda3/lib/python3.6/site-packages/IPython/core/display.py:689: UserWarning:

Consider using IPython.display.IFrame instead



plotly reference: 
file:///Users/libbymurray/Documents/F18/SI370/day7%20(1)/370_07_Visualization_II.html

In [121]:
mdf2= michigan.reset_index()
mdf2['total']= mdf2.sum(axis=1)
mdf2['majority_value']=mdf2['White']/umich['total']
mdf2['minority_value'] = 1 - mdf2['majority_value'] 
mdf2.set_index('Sport')

data = [go.Bar(x=mdf2.Sport,
            y=mdf2.minority_value)]
layout= dict(title='Michigan Race Breakdown by Sport',yaxis= dict(title='Minority Proportion'))


fig = dict(data=data , layout=layout)

py.iplot(fig, filenbame='by_sport_bar')

/Applications/anaconda3/lib/python3.6/site-packages/IPython/core/display.py:689: UserWarning:

Consider using IPython.display.IFrame instead



In [122]:
BTnew_df= BTall_students.reset_index()
BTnew_df['total']= BTnew_df.sum(axis=1)
BTnew_df['majority_value']=BTnew_df['White']/BTdf['total']
BTnew_df['minority_value'] = 1 - BTnew_df['majority_value'] 
BTnew_df.set_index('Sport')

data = [go.Bar(x=BTnew_df.Sport,
            y=BTnew_df.minority_value)]
layout= dict(title='Big-10 Race Breakdown by Sport',yaxis= dict(title='Minority Proportion'))


fig = dict(data=data , layout=layout)

py.iplot(fig, filenbame='Bug10_by_sport_bar')

/Applications/anaconda3/lib/python3.6/site-packages/IPython/core/display.py:689: UserWarning:

Consider using IPython.display.IFrame instead



how to analysis time trends with this data: http://www.ncaa.org/about/resources/research/ncaa-demographics-database


How to overlap barcharts by sport? 

## 5 year span 

In [123]:
timeline_df=pd.read_excel(filename,sheet_name=3)

In [124]:
timeline_df

Sport  white_13  total_13  White_14  total_14  \
0                    Baseball       351       395       337       387   
1            Men's Basketball        76       191        72       189   
2          Women's Basketball        78       173        76       167   
3         Men's Cross Country       204       234       208       235   
4       Women's Cross Country       293       337       303       342   
5                Field Hockey       128       158       124       163   
6                    Football       730      1444       727      1495   
7                  Men's Golf        93       130        84       121   
8                Women's Golf        82       127        81       129   
9            Men's Gymnastics       104       153       109       159   
10         Women's Gymnastics        97       131        99       135   
11                 Ice Hockey       129       165       130       167   
12             Men's Lacrosse       116       141       113       151   
13           Women's Lacrosse        86        99       117       134   
14                     Rowing        80        87       667       771   
15               Men's Soccer       133       192       154       200   
16             Women's Soccer       258       321       267       342   
17                   Softball       211       264       192       253   
18    Men's Swimming & Diving       314       376       289       373   
19  Women's Swimming & Diving       345       422       320       406   
20               Men's Tennis        58       121        71       136   
21             Women's Tennis        53       115        66       122   
22         Men's Indoor Track       404       590       424       615   
23       Women's Indoor Track       467       651       444       646   
24        Men's Outdoor Track       413       601       431       616   
25      Women's Outdoor Track       466       653       447       645   
26                 Volleyball       138       189       161       214   
27                 Water Polo        41        51        42        47   
28                  Wrestling       341       414       324       406   

    white_15  total_15  white_16  total_16  
0        409       468       392       458  
1         73       209        71       218  
2         86       196        62       191  
3        215       255       222       251  
4        333       367       319       365  
5        159       211       167       223  
6        817      1682       803      1704  
7        108       143       117       141  
8         73       134        75       135  
9         95       146       101       149  
10       113       164       113       164  
11       122       164       130       162  
12       202       244       186       232  
13       193       217       189       218  
14       594       742       689       843  
15       191       262       190       272  
16       302       409       309       408  
17       229       300       228       301  
18       285       377       273       364  
19       343       426       351       425  
20        50       120        70       124  
21        68       142        68       137  
22       423       632       390       586  
23       523       743       474       725  
24       438       653       417       623  
25       503       721       473       720  
26       176       246       161       227  
27        37        46        35        43  
28       389       487       371       472

In [125]:
# timelinesorted_sum= ((PACminority_df.sum()).sort_values(ascending =False))
# PACtotal_minority= PACsorted_sum.sum()


# print('total minority student athletes:')
# print (PACtotal_minority)
# PACtotal_students= PACall_students.sum().sum()
# print('total student athletes:')
# print (PACtotal_students )

total_2013=(timeline_df['total_13'].sum())
total_2014=(timeline_df['total_14'].sum())
total_2015=(timeline_df['total_15'].sum())
total_2016=(timeline_df['total_16'].sum())
minority2013= 1-((timeline_df['white_13'].sum())/(timeline_df['total_13'].sum()))
print (minority2013)
minority2014= 1-((timeline_df['White_14'].sum())/(timeline_df['total_14'].sum()))
print (minority2014)
minority2015= 1-((timeline_df['white_15'].sum())/(timeline_df['total_15'].sum()))
print (minority2015)
minority2016= 1-((timeline_df['white_16'].sum())/(timeline_df['total_16'].sum()))
print (minority2016)

0.29535014005602245
0.2956174482899857
0.3078122134604805
0.3156878963330576


In [126]:
newdf= pd.DataFrame(
    {
        "Year":[2013,2014,2015,2016,2017],
        "Minority_%":[minority2013,minority2014,minority2015,minority2016,BTminority_value],
        "Total_students":[total_2013,total_2014,total_2015,total_2016,BTtotal_students]
        
    })
newdf

Year  Minority_%  Total_students
0  2013    0.295350          8925.0
1  2014    0.295617          9766.0
2  2015    0.307812         10906.0
3  2016    0.315688         10881.0
4  2017    0.327075         10918.0

In [127]:
newdf['Year'] =  pd.to_datetime(newdf['Year'], format='%Y')

newdf

Year  Minority_%  Total_students
0 2013-01-01    0.295350          8925.0
1 2014-01-01    0.295617          9766.0
2 2015-01-01    0.307812         10906.0
3 2016-01-01    0.315688         10881.0
4 2017-01-01    0.327075         10918.0

In [128]:
trace_time = go.Scatter(
                x=newdf['Year'],
                y=newdf['Minority_%'],
                name = 'Minority Ratio over 5 years',
                line = dict(color = '#17BECF',width=8),
                opacity = 0.8)

trace_students = go.Scatter(
                x=newdf['Year'],
                y=newdf['Total_students'],
                name = 'Total Student Athlete population change over 5 years',
                line = dict(color = '#7F7F7F'),
                opacity = 0.8)

data = [trace_time] #,trace_low]

layout = dict(title = "Big10 Minority Ratio over 5 years",
              yaxis=dict(title='Minority Ratio',range = [0.25,0.40]),
              xaxis = dict(range = ['2013','2017']))

fig = dict(data=data, layout=layout)
py.iplot(fig, filename = 'Big10-5years-timechart')

/Applications/anaconda3/lib/python3.6/site-packages/IPython/core/display.py:689: UserWarning:

Consider using IPython.display.IFrame instead



### What can we learn from this data?


Michigan’s ratio of minority student athletes is below the national average. 
However, it is on par with the conference average 


Michigan is very close to BigTen, so we can assume Michigan’s diversity is well reflected in Big-Ten’s minority ratio.
While diversity for Michigan and Big-10 are below average, the last 5 years shows Big-Ten’s diversity has a positive progression 


### Further research questions 

* How has diversity by sport changed?
* what does Michigan's 5 year trend look like?
* Look at recruitment spending by sport
* Are sports recruiting from a diverse pool of candidates?
* - explore changes in the home state of athletes 
* Looking at all conferences' 5 year trends 
* Break down Big-10 by schools to see if there is a school affecting the average
